File: pn.c
// PSEUDO RANDOM NUMBER SEQUENCE GENETAION USING LINEAR FEEDBACK SHIFT REGISTER
#include <stdio.h>

int main()
{
    int t[4]= {1, 0, 1, 1};

    // // uncomment for taking the user input
    // printf("Enter 4 binary values (0 or 1) seperated by enter:\n");
    // for (int i = 0; i < 4; i++) {
    //     scanf("%d", &t[i]);
    // }

    printf("input seed: %d%d%d%d", t[0], t[1], t[2], t[3]);
    
    // Generate the PN sequence
    printf("\nPN SEQUENCE: ");
    for (int i = 0; i < 15; i++) {
        int u1 = t[2] ^ t[3]; // XOR operation
        t[3] = t[2];
        t[2] = t[1];
        t[1] = t[0];
        t[0] = u1;
        printf("%d", u1);
    }
    printf("\n");

    return 0;
}

File: pn.py
def generate_pn_sequence():
    # Initialize the shift register with the specified initial values
    x1, x2, x3, x4 = 1, 1, 0, 1  # equivalent to the binary input 1101
    sequence = []

    # Generate the PN sequence for 15 cycles
    for _ in range(15):
        # Calculate the feedback bit (u1) as XOR of x3 and x4
        u1 = x3 ^ x4

        # Only append u1 to match C code output (which only prints u1)
        sequence.append(str(u1))

        # Shift the register values
        x4, x3, x2, x1 = x3, x2, x1, u1

    # Join and print the sequence as a single string
    pn_sequence = ''.join(sequence)
    print("PN Sequence:", pn_sequence)
    return pn_sequence

# Generate and print the PN sequence
result = generate_pn_sequence()